core/src/test/scala/pickling/run/runtime-instantiation-static-picklers.scala

```diff
@@ -0,0 +1,219 @@
+package scala.pickling.runtime
+
+import org.scalatest.FunSuite
+import scala.pickling._, scala.pickling.Defaults._, json._
+
+case class Container[T](value: T)
+
+object RuntimeHelper {
+  /* Instantiate a class for an already generated `Unpickler` */
+  def getInstance[T](className: String): T = {
+    try {
+      val clazz = Class.forName(className)
+      println(s"Create instance of $clazz")
+      clazz.newInstance().asInstanceOf[T]
+    } catch {
+      case ex: Throwable =>
+        scala.concurrent.util.Unsafe.instance.allocateInstance(
+          Class.forName(className)
+        ).asInstanceOf[T]
+    }
+  }
+}
+
+class StaticJavaRuntime extends FunSuite {
+
+  implicit val staticOnly = static.StaticOnly
+
+  test("runtime instantiated pickler should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val containerPickler = implicitly[Pickler[Container[List[Int]]]]
+    val pickled = c.pickle.value
+
+    val picklerClassName = containerPickler.getClass.getName
+    val runtimePickler = RuntimeHelper.getInstance[Pickler[Container[Any]]](picklerClassName)
```

Comment:
@jvican FYI, in "hybrid" mode, you can register static picklers for runtime lookup, but they're still static picklers.

The PicklerRegistry supports an option to REMOVE the runtime pickler generation code.  This is "hybrid-static" picklers, vs. "full hybrid" picklers.

Fully static is where you can know the types of everything when you pickle/unpickle and there's no ambiguity in your ADTs.  "hybrid static" is where you pre-register all known types of picklers, but there is still runtime lookup (and runtime failures) for some of your ADTs.  "hybrid static" allows you to extend your "protocol" while also keeping some static safety.  It's meant to be a blend of the two approaches that lets you sacrifice the minimum of static safety.

In any case, even though I'd argue we should be able to find you an alternative mechanism here, I'll see if I can get the tests working. They *should* work, and you've uncovered another hole in our test suite, where this should be caught/tested.